

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>4. Genome assembly</title>
  

  
  
    <link rel="shortcut icon" href="../_static/favicon.ico"/>
  
  
  

  

  
  
    

  

  
  
    <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  

  
    <link rel="stylesheet" href="../_static/css/seb.css" type="text/css" />
  

  
        <link rel="index" title="Index"
              href="../genindex.html"/>
        <link rel="search" title="Search" href="../search.html"/>
    <link rel="top" title="" href="../index.html"/>
        <link rel="next" title="5. Read mapping" href="../ngs-mapping/index.html"/>
        <link rel="prev" title="3. Quality control" href="../ngs-qc/index.html"/> 

  
  <script src="../_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

   
  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="../index.html" class="icon icon-home"> Genomics Tutorial
          

          
            
            <img src="../_static/icon.png" class="logo" />
          
          </a>

          
            
            
              <div class="version">
                2017.6
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../introduction.html">1. Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-tools/index.html">2. Tool installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-qc/index.html">3. Quality control</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">4. Genome assembly</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#preface">4.1. Preface</a></li>
<li class="toctree-l2"><a class="reference internal" href="#overview">4.2. Overview</a></li>
<li class="toctree-l2"><a class="reference internal" href="#learning-outcomes">4.3. Learning outcomes</a></li>
<li class="toctree-l2"><a class="reference internal" href="#before-we-start">4.4. Before we start</a></li>
<li class="toctree-l2"><a class="reference internal" href="#creating-a-genome-assembly">4.5. Creating a genome assembly</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#installing-the-software">4.5.1. Installing the software</a></li>
<li class="toctree-l3"><a class="reference internal" href="#spades-usage">4.5.2. SPAdes usage</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#assembly-quality-assessment">4.6. Assembly quality assessment</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#assembly-statistics">4.6.1. Assembly statistics</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#compare-the-untrimmed-data">4.7. Compare the untrimmed data</a></li>
<li class="toctree-l2"><a class="reference internal" href="#assemblathon">4.8. Assemblathon</a></li>
<li class="toctree-l2"><a class="reference internal" href="#further-reading">4.9. Further reading</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#background-on-genome-assemblies">4.9.1. Background on Genome Assemblies</a></li>
<li class="toctree-l3"><a class="reference internal" href="#evaluation-of-genome-assembly-software">4.9.2. Evaluation of Genome Assembly Software</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#web-links">4.10. Web links</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-mapping/index.html">5. Read mapping</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-taxonomic-investigation/index.html">6. Taxonomic investigation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-variantcalling/index.html">7. Variant calling</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-annotation/index.html">8. Genome annotation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-orthology/index.html">9. Orthology and Phylogeny</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-voi/index.html">10. Variants-of-interest</a></li>
<li class="toctree-l1"><a class="reference internal" href="../quickref.html">11. Quick command reference</a></li>
<li class="toctree-l1"><a class="reference internal" href="../code.html">12. Coding solutions</a></li>
<li class="toctree-l1"><a class="reference internal" href="../downloads.html">13. Downloads</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Genomics Tutorial</a>
        
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html">Docs</a> &raquo;</li>
        
      <li>4. Genome assembly</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="genome-assembly">
<span id="ngs-assembly"></span><h1>4. Genome assembly<a class="headerlink" href="#genome-assembly" title="Permalink to this headline">¶</a></h1>
<div class="section" id="preface">
<h2>4.1. Preface<a class="headerlink" href="#preface" title="Permalink to this headline">¶</a></h2>
<p>In this section we will use our skill on the command-line interface to create a
genome assembly from sequencing data.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">You will encounter some <strong>To-do</strong> sections at times. Write the solutions and answers into a text-file.</p>
</div>
</div>
<div class="section" id="overview">
<h2>4.2. Overview<a class="headerlink" href="#overview" title="Permalink to this headline">¶</a></h2>
<p>The part of the workflow we will work on in this section can be viewed in <a class="reference internal" href="#fig-workflow-assembly"><span class="std std-numref">Fig. 4.1</span></a>.</p>
<div class="figure" id="id8">
<span id="fig-workflow-assembly"></span><img alt="../_images/workflow1.png" src="../_images/workflow1.png" />
<p class="caption"><span class="caption-number">Fig. 4.1 </span><span class="caption-text">The part of the workflow we will work on in this section marked in red.</span></p>
</div>
</div>
<div class="section" id="learning-outcomes">
<h2>4.3. Learning outcomes<a class="headerlink" href="#learning-outcomes" title="Permalink to this headline">¶</a></h2>
<p>After studying this tutorial you should be able to:</p>
<ol class="arabic simple">
<li>Compute and interpret a whole genome assembly.</li>
<li>Judge the quality of a genome assembly.</li>
</ol>
</div>
<div class="section" id="before-we-start">
<h2>4.4. Before we start<a class="headerlink" href="#before-we-start" title="Permalink to this headline">¶</a></h2>
<p>Lets see how our directory structure looks so far:</p>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="n">cd</span> <span class="o">~/</span><span class="n">analysis</span>
<span class="n">ls</span> <span class="o">-</span><span class="mi">1</span><span class="n">F</span>
</pre></div>
</div>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="n">data</span><span class="o">/</span>
<span class="n">SolexaQA</span><span class="o">/</span>
<span class="n">SolexaQA</span><span class="o">++</span>
<span class="n">trimmed</span><span class="o">/</span>
<span class="n">trimmed</span><span class="o">-</span><span class="n">fastqc</span><span class="o">/</span>
<span class="n">trimmed</span><span class="o">-</span><span class="n">solexaqa</span><span class="o">/</span>
</pre></div>
</div>
</div>
<div class="section" id="creating-a-genome-assembly">
<h2>4.5. Creating a genome assembly<a class="headerlink" href="#creating-a-genome-assembly" title="Permalink to this headline">¶</a></h2>
<p>We want to create a genome assembly for our ancestor.
We are going to use the quality trimmed forward and backward DNA sequences and
use a program called <a class="reference external" href="http://bioinf.spbau.ru/spades">SPAdes</a> to build a genome assembly.</p>
<div class="admonition-todo admonition" id="index-0">
<p class="first admonition-title">Todo</p>
<ol class="last arabic simple">
<li>Discuss briefly why we are using the ancestral sequences to create a
reference genome as opposed to the evolved line.</li>
</ol>
</div>
<div class="section" id="installing-the-software">
<h3>4.5.1. Installing the software<a class="headerlink" href="#installing-the-software" title="Permalink to this headline">¶</a></h3>
<p>We are going to use a program called <a class="reference external" href="http://bioinf.spbau.ru/spades">SPAdes</a> fo assembling our genome.
In a recent evaluation of assembly software, <a class="reference external" href="http://bioinf.spbau.ru/spades">SPAdes</a> was found to be a good
choice for fungal genomes <a class="reference internal" href="#abbas2014" id="id1">[ABBAS2014]</a>.
It is also simple to install and use.</p>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="n">source</span> <span class="n">activate</span> <span class="n">ngs</span>
<span class="n">conda</span> <span class="n">install</span> <span class="n">spades</span>
</pre></div>
</div>
</div>
<div class="section" id="spades-usage">
<h3>4.5.2. SPAdes usage<a class="headerlink" href="#spades-usage" title="Permalink to this headline">¶</a></h3>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="c1"># change to your analysis root folder</span>
<span class="n">cd</span> <span class="o">~/</span><span class="n">analysis</span>

<span class="c1"># first create a output directory for the assemblies</span>
<span class="n">mkdir</span> <span class="n">assembly</span>

<span class="c1"># to get a help for spades and an overview of the parameter type:</span>
<span class="n">spades</span><span class="o">.</span><span class="n">py</span> <span class="o">-</span><span class="n">h</span>
</pre></div>
</div>
<p>The two files we need to submit to <a class="reference external" href="http://bioinf.spbau.ru/spades">SPAdes</a> are two paired-end read files.</p>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="n">spades</span><span class="o">.</span><span class="n">py</span> <span class="o">-</span><span class="n">o</span> <span class="n">assembly</span><span class="o">/</span><span class="n">spades</span><span class="o">-</span><span class="n">default</span><span class="o">/</span> <span class="o">-</span><span class="mi">1</span> <span class="n">trimmed</span><span class="o">/</span><span class="n">ancestor</span><span class="o">-</span><span class="n">R1</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">trimmed</span><span class="o">.</span><span class="n">gz</span> <span class="o">-</span><span class="mi">2</span> <span class="n">trimmed</span><span class="o">/</span><span class="n">ancestor</span><span class="o">-</span><span class="n">R2</span><span class="o">.</span><span class="n">fastq</span><span class="o">.</span><span class="n">trimmed</span><span class="o">.</span><span class="n">gz</span>
</pre></div>
</div>
<div class="admonition-todo admonition" id="index-1">
<p class="first admonition-title">Todo</p>
<ol class="last arabic simple">
<li>Run <a class="reference external" href="http://bioinf.spbau.ru/spades">SPAdes</a> with default parameters on the ancestor</li>
<li>Read in the <a class="reference external" href="http://bioinf.spbau.ru/spades">SPAdes</a> manual about about assembling with 2x150bp reads</li>
<li>Run <a class="reference external" href="http://bioinf.spbau.ru/spades">SPAdes</a> a second time but use the options suggested at the <a class="reference external" href="http://bioinf.spbau.ru/spades">SPAdes</a> manual <a class="reference external" href="http://spades.bioinf.spbau.ru/release3.9.1/manual.html#sec3.4">section 3.4</a> for assembling 2x150bp paired-end reads (are fungi multicellular?). Use a different output directory <code class="docutils literal"><span class="pre">assembly/spades-150</span></code> for this run.</li>
</ol>
</div>
<div class="admonition hint">
<p class="first admonition-title">Hint</p>
<p class="last">Should you not get it right, try the commands in <a class="reference internal" href="../code.html#code-assembly1"><span class="std std-ref">Code: SPAdes assembly (trimmed data)</span></a>.</p>
</div>
</div>
</div>
<div class="section" id="assembly-quality-assessment">
<h2>4.6. Assembly quality assessment<a class="headerlink" href="#assembly-quality-assessment" title="Permalink to this headline">¶</a></h2>
<div class="section" id="assembly-statistics">
<h3>4.6.1. Assembly statistics<a class="headerlink" href="#assembly-statistics" title="Permalink to this headline">¶</a></h3>
<p><a class="reference external" href="http://quast.bioinf.spbau.ru/">Quast</a> (QUality ASsesment Tool) <a class="reference internal" href="#gurevich2013" id="id2">[GUREVICH2013]</a>, evaluates genome assemblies by computing various metrics, including:</p>
<ul class="simple">
<li>N50: length for which the collection of all contigs of that length or
longer covers at least 50% of assembly length</li>
<li>NG50: where length of the reference genome is being covered</li>
<li>NA50 and NGA50: where aligned blocks instead of contigs are taken</li>
<li>missassemblies: misassembled and unaligned contigs or contigs bases</li>
<li>genes and operons covered</li>
</ul>
<p>It is easy with <a class="reference external" href="http://quast.bioinf.spbau.ru/">Quast</a> to compare these measures among several assemblies.
The program can be used on their <a class="reference external" href="http://quast.bioinf.spbau.ru/">website</a>.</p>
<p>We can install it locally with:</p>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="n">source</span> <span class="n">activate</span> <span class="n">ngs</span>
<span class="n">conda</span> <span class="n">install</span> <span class="n">quast</span>
</pre></div>
</div>
<p>Run <a class="reference external" href="http://quast.bioinf.spbau.ru/">Quast</a> with both assembly scaffolds.fasta files to compare the results.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Should you be unable to run <a class="reference external" href="http://bioinf.spbau.ru/spades">SPAdes</a> on the data, you can manually download the assembly from <a class="reference internal" href="../downloads.html#downloads"><span class="std std-ref">Downloads</span></a>. Unarchive and uncompress the files with <code class="docutils literal"><span class="pre">tar</span> <span class="pre">-xvzf</span> <span class="pre">assembly.tar.gz</span></code>.</p>
</div>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span><span class="n">quast</span> <span class="o">-</span><span class="n">o</span> <span class="n">assembly</span><span class="o">/</span><span class="n">quast</span> <span class="n">assembly</span><span class="o">/</span><span class="n">spades</span><span class="o">-</span><span class="n">default</span><span class="o">/</span><span class="n">scaffolds</span><span class="o">.</span><span class="n">fasta</span> <span class="n">assembly</span><span class="o">/</span><span class="n">spades</span><span class="o">-</span><span class="mi">150</span><span class="o">/</span><span class="n">scaffolds</span><span class="o">.</span><span class="n">fasta</span>
</pre></div>
</div>
<div class="admonition-todo admonition" id="index-2">
<p class="first admonition-title">Todo</p>
<ol class="last arabic simple">
<li>Compare the results of <a class="reference external" href="http://quast.bioinf.spbau.ru/">Quast</a> with regards to the two different assemblies.</li>
<li>Which one do you prefer and why?</li>
</ol>
</div>
</div>
</div>
<div class="section" id="compare-the-untrimmed-data">
<h2>4.7. Compare the untrimmed data<a class="headerlink" href="#compare-the-untrimmed-data" title="Permalink to this headline">¶</a></h2>
<div class="admonition-todo admonition" id="index-3">
<p class="first admonition-title">Todo</p>
<ol class="last arabic simple">
<li>To see if our trimming procedure has an influence on our assembly, run the same command you used on the trimmed data on the original untrimmed data.</li>
<li>Run <a class="reference external" href="http://quast.bioinf.spbau.ru/">Quast</a> on the assembly and compare the statistics to the one derived for the trimmed data set. Write down your observations.</li>
</ol>
</div>
<div class="admonition hint">
<p class="first admonition-title">Hint</p>
<p class="last">Should you not get it right, try the commands in <a class="reference internal" href="../code.html#code-assembly2"><span class="std std-ref">Code: SPAdes assembly (original data)</span></a>.</p>
</div>
</div>
<div class="section" id="assemblathon">
<h2>4.8. Assemblathon<a class="headerlink" href="#assemblathon" title="Permalink to this headline">¶</a></h2>
<div class="admonition-todo admonition" id="index-4">
<p class="first admonition-title">Todo</p>
<p class="last">Now that you know the basics for assembling a genome and judging their quality, play with the <a class="reference external" href="http://bioinf.spbau.ru/spades">SPAdes</a> parameters and the <strong>trimmed data</strong> to create the best assembly possible.
We will compare the assemblies to find out who created the best one.</p>
</div>
<div class="admonition-todo admonition" id="index-5">
<p class="first admonition-title">Todo</p>
<ol class="last arabic simple">
<li>Once you have your final assembly, rename your assembly directory int <code class="docutils literal"><span class="pre">spades-final</span></code>, e.g. <code class="docutils literal"><span class="pre">mv</span> <span class="pre">assembly/spades-default</span> <span class="pre">assembly/spades-final</span></code>.</li>
<li>Write down in your notes the command used to create your final assembly.</li>
<li>Write down in your notes the assembly statistics derived through <a class="reference external" href="http://quast.bioinf.spbau.ru/">Quast</a></li>
</ol>
</div>
</div>
<div class="section" id="further-reading">
<h2>4.9. Further reading<a class="headerlink" href="#further-reading" title="Permalink to this headline">¶</a></h2>
<div class="section" id="background-on-genome-assemblies">
<h3>4.9.1. Background on Genome Assemblies<a class="headerlink" href="#background-on-genome-assemblies" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li>How to apply de Bruijn graphs to genome assembly. <a class="reference internal" href="#compeau2011" id="id3">[COMPEAU2011]</a></li>
<li>Sequence assembly demystified. <a class="reference internal" href="#nagarajan2013" id="id4">[NAGARAJAN2013]</a></li>
</ul>
</div>
<div class="section" id="evaluation-of-genome-assembly-software">
<h3>4.9.2. Evaluation of Genome Assembly Software<a class="headerlink" href="#evaluation-of-genome-assembly-software" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li>GAGE: A critical evaluation of genome assemblies and assembly algorithms. <a class="reference internal" href="#salzberg2012" id="id5">[SALZBERG2012]</a></li>
<li>Assessment of de novo assemblers for draft genomes: a case study with fungal genomes. <a class="reference internal" href="#abbas2014" id="id6">[ABBAS2014]</a></li>
</ul>
</div>
</div>
<div class="section" id="web-links">
<h2>4.10. Web links<a class="headerlink" href="#web-links" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li>Lectures for this topic: <a class="reference external" href="https://dx.doi.org/10.6084/m9.figshare.2972323.v1">Genome Assembly: An Introduction</a></li>
<li><a class="reference external" href="http://bioinf.spbau.ru/spades">SPAdes</a></li>
<li><a class="reference external" href="http://quast.bioinf.spbau.ru/">Quast</a></li>
<li><a class="reference external" href="https://rrwick.github.io/Bandage/">Bandage</a> (Bioinformatics Application for Navigating De novo Assembly Graphs Easily) is a program that visualizes a genome assembly as a graph <a class="reference internal" href="#wick2015" id="id7">[WICK2015]</a>.</li>
</ul>
<p class="rubric">References</p>
<table class="docutils citation" frame="void" id="abbas2014" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label">[ABBAS2014]</td><td><em>(<a class="fn-backref" href="#id1">1</a>, <a class="fn-backref" href="#id6">2</a>)</em> Abbas MM, Malluhi QM, Balakrishnan P. Assessment of de novo assemblers for draft genomes: a case study with fungal genomes. <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4290589/">BMC Genomics. 2014;15 Suppl 9:S10.</a> doi: 10.1186/1471-2164-15-S9-S10. Epub 2014 Dec 8.</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="compeau2011" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id3">[COMPEAU2011]</a></td><td>Compeau PE, Pevzner PA, Tesler G. How to apply de Bruijn graphs to genome assembly. <a class="reference external" href="http://dx.doi.org/10.1038/nbt.2023">Nat Biotechnol. 2011 Nov 8;29(11):987-91</a></td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="gurevich2013" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id2">[GUREVICH2013]</a></td><td>Gurevich A, Saveliev V, Vyahhi N and Tesler G. QUAST: quality assessment tool for genome assemblies. <a class="reference external" href="http://bioinformatics.oxfordjournals.org/content/29/8/1072">Bioinformatics 2013, 29(8), 1072-1075</a></td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="nagarajan2013" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id4">[NAGARAJAN2013]</a></td><td>Nagarajan N, Pop M. Sequence assembly demystified. <a class="reference external" href="http://dx.doi.org/10.1038/nrg3367">Nat Rev Genet. 2013 Mar;14(3):157-67</a></td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="salzberg2012" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id5">[SALZBERG2012]</a></td><td>Salzberg SL, Phillippy AM, Zimin A, Puiu D, Magoc T, Koren S, Treangen TJ, Schatz MC, Delcher AL, Roberts M, Marçais G, Pop M, Yorke JA. GAGE: A critical evaluation of genome assemblies and assembly algorithms. <a class="reference external" href="http://genome.cshlp.org/content/22/3/557.full?sid=59ea80f7-b408-4a38-9888-3737bc670876">Genome Res. 2012 Mar;22(3):557-67</a></td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="wick2015" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id7">[WICK2015]</a></td><td>Wick RR, Schultz MB, Zobel J and Holt KE. Bandage: interactive visualization of de novo genome assemblies. <a class="reference external" href="http://bioinformatics.oxfordjournals.org/content/early/2015/07/11/bioinformatics.btv383.long">Bioinformatics 2015, 10.1093/bioinformatics/btv383</a></td></tr>
</tbody>
</table>
</div>
</div>


           </div>
           <div class="articleComments">
            
           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../ngs-mapping/index.html" class="btn btn-neutral float-right" title="5. Read mapping" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="../ngs-qc/index.html" class="btn btn-neutral" title="3. Quality control" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2016-2017, Sebastian Schmeier (https://sschmeier.com).
      Last updated on Jul 30, 2017.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'../',
            VERSION:'2017.6',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true,
            SOURCELINK_SUFFIX: '.txt'
        };
    </script>
      <script type="text/javascript" src="../_static/jquery.js"></script>
      <script type="text/javascript" src="../_static/underscore.js"></script>
      <script type="text/javascript" src="../_static/doctools.js"></script>
      <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>

  

  
  
    <script type="text/javascript" src="../_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>