Introduction
============

This is an introductory tutorial for learning genomics mostly on the command-line.
You will learn how to analyse next-generation sequencing (NGS) data.
The data you will be using is actual research data.
The final aim is to identify the genome variations in evolved lines of wild yeast that can explain the observed biological phenotype.

The workflow
------------

The tutorial workflow is summarised in :numref:`fig-workflow`.

.. _fig-workflow:
.. figure:: /images/workflow.png

   The tutorial will follow this workflow.


Learning outcomes
-----------------

During this tutorial you will learn to:

- Make use of a UNIX-based computer environment
- Check the data quality of an NGS experiment
- Create a genome assembly of the ancestor based on NGS data
- Map NGS reads of evolved lines to the created ancestral reference genome
- Call genome variations/mutations in the evolved lines
- Annotate a newly derived reference genome
- Find variants of interest that may be responsible for the observed evolved phenotype
