

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>6. Taxonomic investigation</title>
  

  
  
    <link rel="shortcut icon" href="../_static/favicon.ico"/>
  
  
  

  

  
  
    

  

  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../_static/css/seb.css" type="text/css" />
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="7. Variant calling" href="../ngs-variantcalling/index.html" />
    <link rel="prev" title="5. Read mapping" href="../ngs-mapping/index.html" />
<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-121349593-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-121349593-1');
</script>


  
  <script src="../_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="../index.html" class="icon icon-home"> Genomics Tutorial
          

          
            
            <img src="../_static/icon.png" class="logo" alt="Logo"/>
          
          </a>

          
            
            
              <div class="version">
                2019.03
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../introduction.html">1. Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-tools/index.html">2. Tool installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-qc/index.html">3. Quality control</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-assembly/index.html">4. Genome assembly</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-mapping/index.html">5. Read mapping</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">6. Taxonomic investigation</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#preface">6.1. Preface</a></li>
<li class="toctree-l2"><a class="reference internal" href="#overview">6.2. Overview</a></li>
<li class="toctree-l2"><a class="reference internal" href="#before-we-start">6.3. Before we start</a></li>
<li class="toctree-l2"><a class="reference internal" href="#kraken2">6.4. Kraken2</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#installation">6.4.1. Installation</a></li>
<li class="toctree-l3"><a class="reference internal" href="#usage">6.4.2. Usage</a></li>
<li class="toctree-l3"><a class="reference internal" href="#investigate-taxa">6.4.3. Investigate taxa</a></li>
<li class="toctree-l3"><a class="reference internal" href="#bracken">6.4.4. Bracken</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#id4">6.4.4.1. Installation</a></li>
<li class="toctree-l4"><a class="reference internal" href="#id5">6.4.4.2. Usage</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#centrifuge">6.5. Centrifuge</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id7">6.5.1. Installation</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id8">6.5.2. Usage</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id9">6.5.3. Investigate taxa</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#centrifuge-report">6.5.3.1. Centrifuge report</a></li>
<li class="toctree-l4"><a class="reference internal" href="#kraken-like-report">6.5.3.2. Kraken-like report</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#visualisation-krona">6.6. Visualisation (Krona)</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id11">6.6.1. Installation</a></li>
<li class="toctree-l3"><a class="reference internal" href="#build-the-taxonomy">6.6.2. Build the taxonomy</a></li>
<li class="toctree-l3"><a class="reference internal" href="#visualise">6.6.3. Visualise</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-variantcalling/index.html">7. Variant calling</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-annotation/index.html">8. Genome annotation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-orthology/index.html">9. Orthology and Phylogeny</a></li>
<li class="toctree-l1"><a class="reference internal" href="../ngs-voi/index.html">10. Variants-of-interest</a></li>
<li class="toctree-l1"><a class="reference internal" href="../quickref.html">11. Quick command reference</a></li>
<li class="toctree-l1"><a class="reference internal" href="../code.html">12. Coding solutions</a></li>
<li class="toctree-l1"><a class="reference internal" href="../downloads.html">13. Downloads</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Genomics Tutorial</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html">Docs</a> &raquo;</li>
        
      <li>6. Taxonomic investigation</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="taxonomic-investigation">
<span id="id1"></span><h1>6. Taxonomic investigation<a class="headerlink" href="#taxonomic-investigation" title="Permalink to this headline">¶</a></h1>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">Since 2020, none of the internal links are functioning. Please use the Dropbox links in the <a class="reference internal" href="../downloads.html#downloads"><span class="std std-ref">Downloads</span></a> section.</p>
</div>
<div class="section" id="preface">
<h2>6.1. Preface<a class="headerlink" href="#preface" title="Permalink to this headline">¶</a></h2>
<p>We want to investigate if there are sequences of other species in our collection of sequenced DNA pieces.
We hope that most of them are from our species that we try to study, i.e. the DNA that we have extracted and amplified.
This might be a way of quality control, e.g. have the samples been contaminated?
Lets investigate if we find sequences from other species in our sequence set.</p>
<p>We will use the tool <a class="reference external" href="https://www.ccb.jhu.edu/software/kraken2/">Kraken2</a> to assign taxonomic classifications to our sequence reads.
Let us see if we can id some sequences from other species.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">You will encounter some <strong>To-do</strong> sections at times. Write the solutions and answers into a text-file.</p>
</div>
</div>
<div class="section" id="overview">
<h2>6.2. Overview<a class="headerlink" href="#overview" title="Permalink to this headline">¶</a></h2>
<p>The part of the workflow we will work on in this section can be viewed in <a class="reference internal" href="#fig-workflow-taxa"><span class="std std-numref">Fig. 6.1</span></a>.</p>
<div class="figure" id="id12">
<span id="fig-workflow-taxa"></span><img alt="../_images/workflow5.png" src="../_images/workflow5.png" />
<p class="caption"><span class="caption-number">Fig. 6.1 </span><span class="caption-text">The part of the workflow we will work on in this section marked in red.</span></p>
</div>
</div>
<div class="section" id="before-we-start">
<h2>6.3. Before we start<a class="headerlink" href="#before-we-start" title="Permalink to this headline">¶</a></h2>
<p>Lets see how our directory structure looks so far:</p>
<div class="code bash highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">cd</span> <span class="o">~/</span><span class="n">analysis</span>
<span class="n">ls</span> <span class="o">-</span><span class="mi">1</span><span class="n">F</span>
</pre></div>
</div>
<div class="code bash highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">assembly</span><span class="o">/</span>
<span class="n">data</span><span class="o">/</span>
<span class="n">mappings</span><span class="o">/</span>
<span class="n">trimmed</span><span class="o">/</span>
<span class="n">trimmed</span><span class="o">-</span><span class="n">fastqc</span><span class="o">/</span>
</pre></div>
</div>
</div>
<div class="section" id="kraken2">
<h2>6.4. Kraken2<a class="headerlink" href="#kraken2" title="Permalink to this headline">¶</a></h2>
<p>We will be using a tool called <a class="reference external" href="https://www.ccb.jhu.edu/software/kraken2/">Kraken2</a> <a class="reference internal" href="#wood2014" id="id2">[WOOD2014]</a>.
This tool uses k-mers to assign a taxonomic labels in form of <a class="reference external" href="https://www.ncbi.nlm.nih.gov/taxonomy">NCBI Taxonomy</a> to the sequence (if possible).
The taxonomic label is assigned based on similar k-mer content of the sequence in question to the k-mer content of reference genome sequence.
The result is a classification of the sequence in question to the most likely taxonomic label.
If the k-mer content is not similar to any genomic sequence in the database used, it will not assign any taxonomic label.</p>
<div class="section" id="installation">
<h3>6.4.1. Installation<a class="headerlink" href="#installation" title="Permalink to this headline">¶</a></h3>
<p>Use conda in the same fashion as before to install <a class="reference external" href="https://www.ccb.jhu.edu/software/kraken2/">Kraken2</a>.
However, we are going to install kraken into its own environment:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>conda create --yes -n kraken kraken2 bracken
conda activate kraken
</pre></div>
</div>
<p>Now we create a directory where we are going to do the analysis and we will change into that directory too.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1"># make sure you are in your analysis root folder</span>
<span class="nb">cd</span> ~/analysis

<span class="c1"># create dir</span>
mkdir kraken
<span class="nb">cd</span> kraken
</pre></div>
</div>
<p>Now we need to create or download a <a class="reference external" href="https://www.ccb.jhu.edu/software/kraken2/">Kraken2</a> database that can be used to assign the taxonomic labels to sequences.
We opt for downloading the pre-build “minikraken2” database from the <a class="reference external" href="https://www.ccb.jhu.edu/software/kraken2/">Kraken2</a> website:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>curl -O ftp://ftp.ccb.jhu.edu/pub/data/kraken2_dbs/minikraken2_v2_8GB_201904_UPDATE.tgz

<span class="c1"># alternatively we can use wget</span>
wget ftp://ftp.ccb.jhu.edu/pub/data/kraken2_dbs/minikraken2_v2_8GB_201904_UPDATE.tgz

<span class="c1"># once the download is finished, we need to extract the archive content:</span>
tar -xvzf minikraken2_v2_8GB_201904_UPDATE.tgz
</pre></div>
</div>
<div class="admonition attention">
<p class="first admonition-title">Attention</p>
<p class="last">Should the download fail. Please find links to alternative locations on the
<a class="reference internal" href="../downloads.html#downloads"><span class="std std-ref">Downloads</span></a> page.</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The “minikraken2” database was created from bacteria, viral and archaea sequences.
What are the implications for us when we are trying to classify our sequences?</p>
</div>
</div>
<div class="section" id="usage">
<h3>6.4.2. Usage<a class="headerlink" href="#usage" title="Permalink to this headline">¶</a></h3>
<p>Now that we have installed <a class="reference external" href="https://www.ccb.jhu.edu/software/kraken2/">Kraken2</a> and downloaded and extracted the minikraken2 database, we can attempt to investigate the sequences we got back from the sequencing provider for other species as the one it should contain.
We call the <a class="reference external" href="https://www.ccb.jhu.edu/software/kraken2/">Kraken2</a> tool and specify the database and fasta-file with the sequences it should use. The general command structure looks like this:</p>
<div class="code bash highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">kraken2</span> <span class="o">--</span><span class="n">use</span><span class="o">-</span><span class="n">names</span> <span class="o">--</span><span class="n">threads</span> <span class="mi">4</span> <span class="o">--</span><span class="n">db</span> <span class="n">PATH_TO_DB_DIR</span> <span class="o">--</span><span class="n">report</span> <span class="n">example</span><span class="o">.</span><span class="n">report</span><span class="o">.</span><span class="n">txt</span> <span class="n">example</span><span class="o">.</span><span class="n">fa</span> <span class="o">&gt;</span> <span class="n">example</span><span class="o">.</span><span class="n">kraken</span>
</pre></div>
</div>
<p>However, we may have fastq-files, so we need to use <code class="docutils literal notranslate"><span class="pre">--fastq-input</span></code> which tells <a class="reference external" href="https://www.ccb.jhu.edu/software/kraken2/">Kraken2</a> that it is dealing with fastq-formated files.
In addition, we are dealing with paired-end data, which we can tell <a class="reference external" href="https://www.ccb.jhu.edu/software/kraken2/">Kraken2</a> with the switch <code class="docutils literal notranslate"><span class="pre">--paired</span></code>.
Here, we are investigating one of the unmapped paired-end read files of the evolved line.</p>
<div class="code bash highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">kraken2</span> <span class="o">--</span><span class="n">use</span><span class="o">-</span><span class="n">names</span> <span class="o">--</span><span class="n">threads</span> <span class="mi">4</span> <span class="o">--</span><span class="n">db</span> <span class="n">minikraken2_v2_8GB_201904_UPDATE</span> <span class="o">--</span><span class="n">fastq</span><span class="o">-</span><span class="nb">input</span> <span class="o">--</span><span class="n">report</span> <span class="n">evolved</span><span class="o">-</span><span class="mi">6</span> <span class="o">--</span><span class="n">paired</span> <span class="o">../</span><span class="n">mappings</span><span class="o">/</span><span class="n">evolved</span><span class="o">-</span><span class="mf">6.</span><span class="n">sorted</span><span class="o">.</span><span class="n">unmapped</span><span class="o">.</span><span class="n">R1</span><span class="o">.</span><span class="n">fastq</span> <span class="o">../</span><span class="n">mappings</span><span class="o">/</span><span class="n">evolved</span><span class="o">-</span><span class="mf">6.</span><span class="n">sorted</span><span class="o">.</span><span class="n">unmapped</span><span class="o">.</span><span class="n">R2</span><span class="o">.</span><span class="n">fastq</span> <span class="o">&gt;</span> <span class="n">evolved</span><span class="o">-</span><span class="mf">6.</span><span class="n">kraken</span>
</pre></div>
</div>
<p>This classification may take a while, depending on how many sequences we are going to classify.
The resulting content of the file “evolved-6.kraken” looks similar to the following example:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">C</span>       <span class="mi">7001326</span><span class="n">F</span><span class="p">:</span><span class="mi">121</span><span class="p">:</span><span class="n">CBVVLANXX</span><span class="p">:</span><span class="mi">1</span><span class="p">:</span><span class="mi">1105</span><span class="p">:</span><span class="mi">2240</span><span class="p">:</span><span class="mi">12640</span>        <span class="mi">816</span>     <span class="mi">251</span>     <span class="mi">816</span><span class="p">:</span><span class="mi">9</span> <span class="mi">171549</span><span class="p">:</span><span class="mi">5</span> <span class="mi">816</span><span class="p">:</span><span class="mi">5</span> <span class="mi">171549</span><span class="p">:</span><span class="mi">3</span> <span class="mi">2</span><span class="p">:</span><span class="mi">2</span> <span class="mi">816</span><span class="p">:</span><span class="mi">5</span> <span class="mi">171549</span><span class="p">:</span><span class="mi">4</span> <span class="mi">816</span><span class="p">:</span><span class="mi">34</span> <span class="mi">171549</span><span class="p">:</span><span class="mi">8</span> <span class="mi">816</span><span class="p">:</span><span class="mi">4</span> <span class="mi">171549</span><span class="p">:</span><span class="mi">2</span> <span class="mi">816</span><span class="p">:</span><span class="mi">10</span> <span class="n">A</span><span class="p">:</span><span class="mi">35</span> <span class="mi">816</span><span class="p">:</span><span class="mi">10</span> <span class="mi">171549</span><span class="p">:</span><span class="mi">2</span> <span class="mi">816</span><span class="p">:</span><span class="mi">4</span> <span class="mi">171549</span><span class="p">:</span><span class="mi">8</span> <span class="mi">816</span><span class="p">:</span><span class="mi">34</span> <span class="mi">171549</span><span class="p">:</span><span class="mi">4</span> <span class="mi">816</span><span class="p">:</span><span class="mi">5</span> <span class="mi">2</span><span class="p">:</span><span class="mi">2</span> <span class="mi">171549</span><span class="p">:</span><span class="mi">3</span> <span class="mi">816</span><span class="p">:</span><span class="mi">5</span> <span class="mi">171549</span><span class="p">:</span><span class="mi">5</span> <span class="mi">816</span><span class="p">:</span><span class="mi">9</span>
<span class="n">C</span>       <span class="mi">7001326</span><span class="n">F</span><span class="p">:</span><span class="mi">121</span><span class="p">:</span><span class="n">CBVVLANXX</span><span class="p">:</span><span class="mi">1</span><span class="p">:</span><span class="mi">1105</span><span class="p">:</span><span class="mi">3487</span><span class="p">:</span><span class="mi">12536</span>        <span class="mi">1339337</span> <span class="mi">202</span>     <span class="mi">1339337</span><span class="p">:</span><span class="mi">67</span> <span class="n">A</span><span class="p">:</span><span class="mi">35</span> <span class="mi">1339337</span><span class="p">:</span><span class="mi">66</span>
<span class="n">U</span>       <span class="mi">7001326</span><span class="n">F</span><span class="p">:</span><span class="mi">121</span><span class="p">:</span><span class="n">CBVVLANXX</span><span class="p">:</span><span class="mi">1</span><span class="p">:</span><span class="mi">1105</span><span class="p">:</span><span class="mi">5188</span><span class="p">:</span><span class="mi">12504</span>        <span class="mi">0</span>       <span class="mi">251</span>     <span class="mi">0</span><span class="p">:</span><span class="mi">91</span> <span class="n">A</span><span class="p">:</span><span class="mi">35</span> <span class="mi">0</span><span class="p">:</span><span class="mi">91</span>
<span class="n">U</span>       <span class="mi">7001326</span><span class="n">F</span><span class="p">:</span><span class="mi">121</span><span class="p">:</span><span class="n">CBVVLANXX</span><span class="p">:</span><span class="mi">1</span><span class="p">:</span><span class="mi">1105</span><span class="p">:</span><span class="mi">11030</span><span class="p">:</span><span class="mi">12689</span>       <span class="mi">0</span>       <span class="mi">251</span>     <span class="mi">0</span><span class="p">:</span><span class="mi">91</span> <span class="n">A</span><span class="p">:</span><span class="mi">35</span> <span class="mi">0</span><span class="p">:</span><span class="mi">91</span>
<span class="n">U</span>       <span class="mi">7001326</span><span class="n">F</span><span class="p">:</span><span class="mi">121</span><span class="p">:</span><span class="n">CBVVLANXX</span><span class="p">:</span><span class="mi">1</span><span class="p">:</span><span class="mi">1105</span><span class="p">:</span><span class="mi">7157</span><span class="p">:</span><span class="mi">12806</span>        <span class="mi">0</span>       <span class="mi">206</span>     <span class="mi">0</span><span class="p">:</span><span class="mi">69</span> <span class="n">A</span><span class="p">:</span><span class="mi">35</span> <span class="mi">0</span><span class="p">:</span><span class="mi">68</span>
</pre></div>
</div>
<p>Each sequence classified by <a class="reference external" href="https://www.ccb.jhu.edu/software/kraken2/">Kraken2</a> results in a single line of output.
Output lines contain five tab-delimited fields; from left to right, they are:</p>
<ol class="arabic simple">
<li><code class="docutils literal notranslate"><span class="pre">C/U</span></code>: one letter code indicating that the sequence was either classified or unclassified.</li>
<li>The sequence ID, obtained from the FASTA/FASTQ header.</li>
<li>The taxonomy ID <a class="reference external" href="https://www.ccb.jhu.edu/software/kraken2/">Kraken2</a> used to label the sequence; this is <strong>0</strong> if the sequence
is unclassified and otherwise should be the <a class="reference external" href="https://www.ncbi.nlm.nih.gov/taxonomy">NCBI Taxonomy</a> identifier.</li>
<li>The length of the sequence in bp.</li>
<li>A space-delimited list indicating the lowest common ancestor (in the
taxonomic tree) mapping of each k-mer in the sequence.
For example, <code class="docutils literal notranslate"><span class="pre">562:13</span> <span class="pre">561:4</span> <span class="pre">A:31</span> <span class="pre">0:1</span> <span class="pre">562:3</span></code> would indicate that:<ul>
<li>the first 13 k-mers mapped to taxonomy ID #562</li>
<li>the next 4 k-mers mapped to taxonomy ID #561</li>
<li>the next 31 k-mers contained an ambiguous nucleotide</li>
<li>the next k-mer was not in the database</li>
<li>the last 3 k-mers mapped to taxonomy ID #562</li>
</ul>
</li>
</ol>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The <a class="reference external" href="https://www.ccb.jhu.edu/software/kraken2/">Kraken2</a> manual can be accessed <a class="reference external" href="https://www.ccb.jhu.edu/software/kraken2/index.shtml?t=manual">here</a>.</p>
</div>
</div>
<div class="section" id="investigate-taxa">
<h3>6.4.3. Investigate taxa<a class="headerlink" href="#investigate-taxa" title="Permalink to this headline">¶</a></h3>
<p>We can use the webpage <a class="reference external" href="https://www.ncbi.nlm.nih.gov/Taxonomy/TaxIdentifier/tax_identifier.cgi">NCBI TaxIdentifier</a> to quickly get the names to the taxonomy identifier.
However, this is impractical as we are dealing potentially with many sequences.
<a class="reference external" href="https://www.ccb.jhu.edu/software/kraken2/">Kraken2</a> has some scripts that help us understand our results better.</p>
<p>Because we used the <a class="reference external" href="https://www.ccb.jhu.edu/software/kraken2/">Kraken2</a> switch <code class="docutils literal notranslate"><span class="pre">--report</span> <span class="pre">FILE</span></code>, we have got also a sample-wide report of all taxa found.
This is much better to get an overview what was found.</p>
<p>The first few lines of an example report are shown below.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="mf">83.56</span>  <span class="mi">514312</span>  <span class="mi">514312</span>  <span class="n">U</span>       <span class="mi">0</span>       <span class="n">unclassified</span>
<span class="mf">16.44</span>  <span class="mi">101180</span>  <span class="mi">0</span>       <span class="n">R</span>       <span class="mi">1</span>       <span class="n">root</span>
<span class="mf">16.44</span>  <span class="mi">101180</span>  <span class="mi">0</span>       <span class="n">R1</span>      <span class="mi">131567</span>    <span class="n">cellular</span> <span class="n">organisms</span>
<span class="mf">16.44</span>  <span class="mi">101180</span>  <span class="mi">2775</span>    <span class="n">D</span>       <span class="mi">2</span>           <span class="n">Bacteria</span>
<span class="mf">13.99</span>  <span class="mi">86114</span>   <span class="mi">1</span>       <span class="n">D1</span>      <span class="mi">1783270</span>       <span class="n">FCB</span> <span class="n">group</span>
<span class="mf">13.99</span>  <span class="mi">86112</span>   <span class="mi">0</span>       <span class="n">D2</span>      <span class="mi">68336</span>           <span class="n">Bacteroidetes</span><span class="o">/</span><span class="n">Chlorobi</span> <span class="n">group</span>
<span class="mf">13.99</span>  <span class="mi">86103</span>   <span class="mi">8</span>       <span class="n">P</span>       <span class="mi">976</span>               <span class="n">Bacteroidetes</span>
<span class="mf">13.94</span>  <span class="mi">85798</span>   <span class="mi">2</span>       <span class="n">C</span>       <span class="mi">200643</span>              <span class="n">Bacteroidia</span>
<span class="mf">13.94</span>  <span class="mi">85789</span>   <span class="mi">19</span>      <span class="n">O</span>       <span class="mi">171549</span>                <span class="n">Bacteroidales</span>
<span class="mf">13.87</span>  <span class="mi">85392</span>   <span class="mi">0</span>       <span class="n">F</span>       <span class="mi">815</span>                     <span class="n">Bacteroidaceae</span>
</pre></div>
</div>
<p>The output of kraken-report is tab-delimited, with one line per taxon.
The fields of the output, from left-to-right, are as follows:</p>
<ol class="arabic simple">
<li><strong>Percentage</strong> of reads covered by the clade rooted at this taxon</li>
<li><strong>Number of reads</strong> covered by the clade rooted at this taxon</li>
<li><strong>Number of reads</strong> assigned directly to this taxon</li>
<li>A rank code, indicating <strong>(U)nclassified, (D)omain, (K)ingdom, (P)hylum, (C)lass, (O)rder, (F)amily, (G)enus, or (S)pecies</strong>. All other ranks are simply <strong>“-“</strong>.</li>
<li><a class="reference external" href="https://www.ncbi.nlm.nih.gov/taxonomy">NCBI Taxonomy</a> ID</li>
<li>The indented scientific name</li>
</ol>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">If you want to compare the taxa content of different samples to another, one can create a report whose structure is always the same for all samples, disregarding which taxa are found (obviously the percentages and numbers will be different).</p>
</div>
<p>We can cerate such a report using the option <code class="docutils literal notranslate"><span class="pre">--report-zero-counts</span></code> which will print out all taxa (instead of only those found).
We then sort the taxa according to taxa-ids (column 5), e.g. <code class="docutils literal notranslate"><span class="pre">sort</span> <span class="pre">-n</span> <span class="pre">-k5</span></code>.</p>
<p>The report is not ordered according to taxa ids and contains all taxa in the database, even if they have not been found in our sample and are thus zero.
The columns are the same as in the former report, however, we have more rows and they are now differently sorted, according to the <a class="reference external" href="https://www.ncbi.nlm.nih.gov/taxonomy">NCBI Taxonomy</a> id.</p>
</div>
<div class="section" id="bracken">
<h3>6.4.4. Bracken<a class="headerlink" href="#bracken" title="Permalink to this headline">¶</a></h3>
<p><a class="reference external" href="https://ccb.jhu.edu/software/bracken/index.shtml">Bracken</a> stands for Bayesian Re-estimation of Abundance with KrakEN, and is a statistical method that computes the abundance of species in DNA sequences from a metagenomics sample <a class="reference internal" href="#lu2017" id="id3">[LU2017]</a>.
<a class="reference external" href="https://ccb.jhu.edu/software/bracken/index.shtml">Bracken</a> uses the taxonomy labels assigned by <a class="reference external" href="https://www.ccb.jhu.edu/software/kraken2/">Kraken2</a> (see above) to estimate the number of reads originating from each species present in a sample.
<a class="reference external" href="https://ccb.jhu.edu/software/bracken/index.shtml">Bracken</a> classifies reads to the best matching location in the taxonomic tree, but does not estimate abundances of species.
Combined with the Kraken classifier, <a class="reference external" href="https://ccb.jhu.edu/software/bracken/index.shtml">Bracken</a> will produces more accurate species- and genus-level abundance estimates than <a class="reference external" href="https://www.ccb.jhu.edu/software/kraken2/">Kraken2</a> alone.</p>
<p>The use of <a class="reference external" href="https://ccb.jhu.edu/software/bracken/index.shtml">Bracken</a> subsequent to <a class="reference external" href="https://www.ccb.jhu.edu/software/kraken2/">Kraken2</a> is optional but might improve on the <a class="reference external" href="https://www.ccb.jhu.edu/software/kraken2/">Kraken2</a> results.</p>
<div class="section" id="id4">
<h4>6.4.4.1. Installation<a class="headerlink" href="#id4" title="Permalink to this headline">¶</a></h4>
<p>We installed <a class="reference external" href="https://ccb.jhu.edu/software/bracken/index.shtml">Bracken</a> already together with <a class="reference external" href="https://www.ccb.jhu.edu/software/kraken2/">Kraken2</a> above, so it should be ready to be used.
We also downloaded the <a class="reference external" href="https://ccb.jhu.edu/software/bracken/index.shtml">Bracken</a> files together with the minikraken2 database above, so we are good to go.</p>
</div>
<div class="section" id="id5">
<h4>6.4.4.2. Usage<a class="headerlink" href="#id5" title="Permalink to this headline">¶</a></h4>
<p>Now, we can use <a class="reference external" href="https://ccb.jhu.edu/software/bracken/index.shtml">Bracken</a> on the <a class="reference external" href="https://www.ccb.jhu.edu/software/kraken2/">Kraken2</a> results to improve them.</p>
<p>The general structure of the <a class="reference external" href="https://ccb.jhu.edu/software/bracken/index.shtml">Bracken</a> command look like this:</p>
<div class="code bash highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">bracken</span> <span class="o">-</span><span class="n">d</span> <span class="n">PATH_TO_DB_DIR</span> <span class="o">-</span><span class="n">i</span> <span class="n">kraken2</span><span class="o">.</span><span class="n">report</span> <span class="o">-</span><span class="n">o</span> <span class="n">bracken</span><span class="o">.</span><span class="n">species</span><span class="o">.</span><span class="n">txt</span> <span class="o">-</span><span class="n">l</span> <span class="n">S</span>
</pre></div>
</div>
<ul class="simple">
<li><code class="docutils literal notranslate"><span class="pre">-l</span> <span class="pre">S</span></code>: denotes the level we want to look at. <code class="docutils literal notranslate"><span class="pre">S</span></code> stands for species but other levels are available.</li>
<li><code class="docutils literal notranslate"><span class="pre">-d</span> <span class="pre">PATH_TO_DB_DIR</span></code>: specifies the path to the <a class="reference external" href="https://www.ccb.jhu.edu/software/kraken2/">Kraken2</a> database that should be used.</li>
</ul>
<p>Let us apply <a class="reference external" href="https://ccb.jhu.edu/software/bracken/index.shtml">Bracken</a> to the example above:</p>
<div class="code bash highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">bracken</span> <span class="o">-</span><span class="n">d</span> <span class="n">minikraken2_v2_8GB_201904_UPDATE</span> <span class="o">-</span><span class="n">i</span> <span class="n">evolved</span><span class="o">-</span><span class="mf">6.</span><span class="n">kraken</span> <span class="o">-</span><span class="n">l</span> <span class="n">S</span> <span class="o">-</span><span class="n">o</span> <span class="n">evolved</span><span class="o">-</span><span class="mf">6.</span><span class="n">bracken</span>
</pre></div>
</div>
<p>The species-focused result-table looks similar to this:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">name</span>    <span class="n">taxonomy_id</span>     <span class="n">taxonomy_lvl</span>    <span class="n">kraken_assigned_reads</span>   <span class="n">added_reads</span>     <span class="n">new_est_reads</span>   <span class="n">fraction_total_reads</span>
<span class="n">Streptococcus</span> <span class="n">sp</span><span class="o">.</span> <span class="n">oral</span> <span class="n">taxon</span> <span class="mi">431</span>        <span class="mi">712633</span>  <span class="n">S</span>       <span class="mi">2</span>       <span class="mi">0</span>       <span class="mi">2</span>       <span class="mf">0.00001</span>
<span class="n">Neorhizobium</span> <span class="n">sp</span><span class="o">.</span> <span class="n">NCHU2750</span>       <span class="mi">1825976</span> <span class="n">S</span>       <span class="mi">0</span>       <span class="mi">0</span>       <span class="mi">0</span>       <span class="mf">0.00000</span>
<span class="n">Pseudomonas</span> <span class="n">sp</span><span class="o">.</span> <span class="n">MT</span><span class="o">-</span><span class="mi">1</span>    <span class="mi">150396</span>  <span class="n">S</span>       <span class="mi">0</span>       <span class="mi">0</span>       <span class="mi">0</span>       <span class="mf">0.00000</span>
<span class="n">Ahniella</span> <span class="n">affigens</span>       <span class="mi">2021234</span> <span class="n">S</span>       <span class="mi">1</span>       <span class="mi">0</span>       <span class="mi">1</span>       <span class="mf">0.00000</span>
<span class="n">Sinorhizobium</span> <span class="n">sp</span><span class="o">.</span> <span class="n">CCBAU</span> <span class="mi">05631</span>   <span class="mi">794846</span>  <span class="n">S</span>       <span class="mi">0</span>       <span class="mi">0</span>       <span class="mi">0</span>       <span class="mf">0.00000</span>
<span class="n">Cohnella</span> <span class="n">sp</span><span class="o">.</span> <span class="mi">18</span><span class="n">JY8</span><span class="o">-</span><span class="mi">7</span>    <span class="mi">2480923</span> <span class="n">S</span>       <span class="mi">1</span>       <span class="mi">0</span>       <span class="mi">1</span>       <span class="mf">0.00000</span>
<span class="n">Bacillus</span> <span class="n">velezensis</span>     <span class="mi">492670</span>  <span class="n">S</span>       <span class="mi">4</span>       <span class="mi">4</span>       <span class="mi">8</span>       <span class="mf">0.00002</span>
<span class="n">Actinoplanes</span> <span class="n">missouriensis</span>      <span class="mi">1866</span>    <span class="n">S</span>       <span class="mi">2</span>       <span class="mi">8</span>       <span class="mi">10</span>      <span class="mf">0.00002</span>
</pre></div>
</div>
<p>The important column is the <code class="docutils literal notranslate"><span class="pre">new_est_reads</span></code>, which gives the newly estimated reads.</p>
</div>
</div>
</div>
<div class="section" id="centrifuge">
<h2>6.5. Centrifuge<a class="headerlink" href="#centrifuge" title="Permalink to this headline">¶</a></h2>
<p>We can also use another tool by the same group called <a class="reference external" href="http://www.ccb.jhu.edu/software/centrifuge/index.shtml">Centrifuge</a> <a class="reference internal" href="#kim2017" id="id6">[KIM2017]</a>.
This tool uses a novel indexing scheme based on the Burrows-Wheeler transform (BWT) and the Ferragina-Manzini (FM) index, optimized specifically for the metagenomic classification problem to assign a taxonomic labels in form of <a class="reference external" href="https://www.ncbi.nlm.nih.gov/taxonomy">NCBI Taxonomy</a> to the sequence (if possible).
The result is a classification of the sequence in question to the most likely taxonomic label.
If the search sequence is not similar to any genomic sequence in the database used, it will not assign any taxonomic label.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">I would normally use <a class="reference external" href="https://www.ccb.jhu.edu/software/kraken2/">Kraken2</a> and only prefer <a class="reference external" href="http://www.ccb.jhu.edu/software/centrifuge/index.shtml">Centrifuge</a> if memory and/or speed are an issue .</p>
</div>
<div class="section" id="id7">
<h3>6.5.1. Installation<a class="headerlink" href="#id7" title="Permalink to this headline">¶</a></h3>
<p>Use conda in the same fashion as before to install <a class="reference external" href="http://www.ccb.jhu.edu/software/centrifuge/index.shtml">Centrifuge</a>:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>conda create --yes -n centrifuge centrifuge
conda activate centrifuge
</pre></div>
</div>
<p>Now we create a directory where we are going to do the analysis and we will change into that directory too.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1"># make sure you are in your analysis root folder</span>
<span class="nb">cd</span> ~/analysis

<span class="c1"># create dir</span>
mkdir centrifuge
<span class="nb">cd</span> centrifuge
</pre></div>
</div>
<p>Now we need to create or download a <a class="reference external" href="http://www.ccb.jhu.edu/software/centrifuge/index.shtml">Centrifuge</a> database that can be used to assign the taxonomic labels to sequences.
We opt for downloading the pre-build database from the <a class="reference external" href="http://www.ccb.jhu.edu/software/centrifuge/index.shtml">Centrifuge</a> website:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>curl -O ftp://ftp.ccb.jhu.edu/pub/infphilo/centrifuge/data/p_compressed+h+v.tar.gz

<span class="c1"># alternatively we can use wget</span>
wget ftp://ftp.ccb.jhu.edu/pub/infphilo/centrifuge/data/p_compressed+h+v.tar.gz

<span class="c1"># once the download is finished, we need to extract the archive content</span>
<span class="c1"># It will extract a few files from the archive and may take a moment to finish.</span>
tar -xvzf p_compressed+h+v.tar.gz
</pre></div>
</div>
<div class="admonition attention">
<p class="first admonition-title">Attention</p>
<p class="last">Should the download fail. Please find links to alternative locations on the
<a class="reference internal" href="../downloads.html#downloads"><span class="std std-ref">Downloads</span></a> page.</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The database we will be using was created from bacteria and archaea sequences only.
What are the implications for us when we are trying to classify our sequences?</p>
</div>
</div>
<div class="section" id="id8">
<h3>6.5.2. Usage<a class="headerlink" href="#id8" title="Permalink to this headline">¶</a></h3>
<p>Now that we have installed <a class="reference external" href="http://www.ccb.jhu.edu/software/centrifuge/index.shtml">Centrifuge</a> and downloaded and extracted the pre-build database, we can attempt to investigate the sequences we got back from the sequencing provider for other species as the one it should contain.
We call the <a class="reference external" href="http://www.ccb.jhu.edu/software/centrifuge/index.shtml">Centrifuge</a> tool and specify the database and fasta-file with the sequences it should use. The general command structure looks like this:</p>
<div class="code bash highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">centrifuge</span> <span class="o">-</span><span class="n">x</span> <span class="n">p_compressed</span><span class="o">+</span><span class="n">h</span><span class="o">+</span><span class="n">v</span> <span class="o">-</span><span class="n">U</span> <span class="n">example</span><span class="o">.</span><span class="n">fa</span> <span class="o">--</span><span class="n">report</span><span class="o">-</span><span class="n">file</span> <span class="n">report</span><span class="o">.</span><span class="n">txt</span> <span class="o">-</span><span class="n">S</span> <span class="n">results</span><span class="o">.</span><span class="n">txt</span>
</pre></div>
</div>
<p>However, if we do not have fastq-files we may have to use the  <code class="docutils literal notranslate"><span class="pre">-f</span></code> option, which tells <a class="reference external" href="http://www.ccb.jhu.edu/software/centrifuge/index.shtml">Centrifuge</a> that it is dealing with a fasta-formated file.
Here, we are investigating one of the unmapped paired-end read files of the evolved line.</p>
<div class="code bash highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">centrifuge</span> <span class="o">-</span><span class="n">x</span> <span class="n">p_compressed</span><span class="o">+</span><span class="n">h</span><span class="o">+</span><span class="n">v</span> <span class="o">-</span><span class="n">U</span> <span class="o">../</span><span class="n">mappings</span><span class="o">/</span><span class="n">evolved</span><span class="o">-</span><span class="mf">6.</span><span class="n">sorted</span><span class="o">.</span><span class="n">unmapped</span><span class="o">.</span><span class="n">R1</span><span class="o">.</span><span class="n">fastq</span> <span class="o">--</span><span class="n">report</span><span class="o">-</span><span class="n">file</span> <span class="n">evolved</span><span class="o">-</span><span class="mi">6</span><span class="o">-</span><span class="n">R1</span><span class="o">-</span><span class="n">report</span><span class="o">.</span><span class="n">txt</span> <span class="o">-</span><span class="n">S</span> <span class="n">evolved</span><span class="o">-</span><span class="mi">6</span><span class="o">-</span><span class="n">R1</span><span class="o">-</span><span class="n">results</span><span class="o">.</span><span class="n">txt</span>
</pre></div>
</div>
<p>This classification may take a moment, depending on how many sequences we are going to classify.
The resulting content of the file <code class="docutils literal notranslate"><span class="pre">evolved-6-R1-results.txt</span></code> looks similar to the following example:</p>
<pre class="literal-block">
readID  seqID   taxID   score   2ndBestScore    hitLength       queryLength     numMatches
M02810:197:000000000-AV55U:1:1101:15316:8461    cid|1747        1747    1892    0       103     135     1
M02810:197:000000000-AV55U:1:1101:15563:3249    cid|161879      161879  18496   0       151     151     1
M02810:197:000000000-AV55U:1:1101:19743:5166    cid|564 564     10404   10404   117     151     2
M02810:197:000000000-AV55U:1:1101:19743:5166    cid|562 562     10404   10404   117     151     2

</pre>
<p>Each sequence classified by <a class="reference external" href="http://www.ccb.jhu.edu/software/centrifuge/index.shtml">Centrifuge</a> results in a single line of output.
Output lines contain eight tab-delimited fields; from left to right, they are according to the <a class="reference external" href="http://www.ccb.jhu.edu/software/centrifuge/index.shtml">Centrifuge</a> website:</p>
<ol class="arabic simple">
<li>The read ID from a raw sequencing read.</li>
<li>The sequence ID of the genomic sequence, where the read is classified.</li>
<li>The taxonomic ID of the genomic sequence in the second column.</li>
<li>The score for the classification, which is the weighted sum of hits.</li>
<li>The score for the next best classification.</li>
<li>A pair of two numbers: (1) an approximate number of base pairs of the read that match the genomic sequence and (2) the length of a read or the combined length of mate pairs.</li>
<li>A pair of two numbers: (1) an approximate number of base pairs of the read that match the genomic sequence and (2) the length of a read or the combined length of mate pairs.</li>
<li>The  number of classifications for this read, indicating how many assignments were made.</li>
</ol>
</div>
<div class="section" id="id9">
<h3>6.5.3. Investigate taxa<a class="headerlink" href="#id9" title="Permalink to this headline">¶</a></h3>
<div class="section" id="centrifuge-report">
<h4>6.5.3.1. Centrifuge report<a class="headerlink" href="#centrifuge-report" title="Permalink to this headline">¶</a></h4>
<p>The command above creates a <a class="reference external" href="http://www.ccb.jhu.edu/software/centrifuge/index.shtml">Centrifuge</a> report automatically for us.
It contains an overview of the identified taxa and their abundances in your supplied sequences (normalised to genomic length):</p>
<pre class="literal-block">
name    taxID   taxRank genomeSize      numReads        numUniqueReads  abundance
Pseudomonas aeruginosa  287     species 22457305        1       0       0.0
Pseudomonas fluorescens 294     species 14826544        1       1       0.0
Pseudomonas putida      303     species 6888188 1       1       0.0
Ralstonia pickettii     329     species 6378979 3       2       0.0
Pseudomonas pseudoalcaligenes   330     species 4691662 1       1       0.0171143

</pre>
<p>Each line contains seven tab-delimited fields; from left to right, they are according to the <a class="reference external" href="http://www.ccb.jhu.edu/software/centrifuge/index.shtml">Centrifuge</a> website:</p>
<ol class="arabic simple">
<li>The name of a genome, or the name corresponding to a taxonomic ID (the second column) at a rank higher than the strain.</li>
<li>The taxonomic ID.</li>
<li>The taxonomic rank.</li>
<li>The length of the genome sequence.</li>
<li>The number of reads classified to this genomic sequence including multi-classified reads.</li>
<li>The number of reads uniquely classified to this genomic sequence.</li>
<li>The proportion of this genome normalized by its genomic length.</li>
</ol>
</div>
<div class="section" id="kraken-like-report">
<h4>6.5.3.2. Kraken-like report<a class="headerlink" href="#kraken-like-report" title="Permalink to this headline">¶</a></h4>
<p>If we would like to generate a report as generated with the former tool <a class="reference external" href="https://www.ccb.jhu.edu/software/kraken2/">Kraken2</a>, we can do it like this:</p>
<div class="code highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">centrifuge</span><span class="o">-</span><span class="n">kreport</span> <span class="o">-</span><span class="n">x</span> <span class="n">p_compressed</span><span class="o">+</span><span class="n">h</span><span class="o">+</span><span class="n">v</span> <span class="n">evolved</span><span class="o">-</span><span class="mi">6</span><span class="o">-</span><span class="n">R1</span><span class="o">-</span><span class="n">results</span><span class="o">.</span><span class="n">txt</span> <span class="o">&gt;</span> <span class="n">evolved</span><span class="o">-</span><span class="mi">6</span><span class="o">-</span><span class="n">R1</span><span class="o">-</span><span class="n">kreport</span><span class="o">.</span><span class="n">txt</span>
</pre></div>
</div>
<pre class="literal-block">
  0.00  0       0       U       0       unclassified
 78.74  163     0       -       1       root
 78.74  163     0       -       131567    cellular organisms
 78.74  163     0       D       2           Bacteria
 54.67  113     0       P       1224          Proteobacteria
 36.60  75      0       C       1236            Gammaproteobacteria
 31.18  64      0       O       91347             Enterobacterales
 30.96  64      0       F       543                 Enterobacteriaceae
 23.89  49      0       G       561                   Escherichia
 23.37  48      48      S       562                     Escherichia coli
  0.40  0       0       S       564                     Escherichia fergusonii
  0.12  0       0       S       208962                  Escherichia albertii
  3.26  6       0       G       570                   Klebsiella
  3.14  6       6       S       573                     Klebsiella pneumoniae
  0.12  0       0       S       548                     [Enterobacter] aerogenes
  2.92  6       0       G       620                   Shigella
  1.13  2       2       S       623                     Shigella flexneri
  0.82  1       1       S       624                     Shigella sonnei
  0.50  1       1       S       1813821                 Shigella sp. PAMC 28760
  0.38  0       0       S       621                     Shigella boydii

</pre>
<p>This gives a similar (not the same) report as the <a class="reference external" href="https://www.ccb.jhu.edu/software/kraken2/">Kraken2</a> tool. The report is tab-delimited, with one line per taxon. The fields of the output, from left-to-right, are as follows:</p>
<ol class="arabic simple">
<li>Percentage of reads covered by the clade rooted at this taxon</li>
<li>Number of reads covered by the clade rooted at this taxon</li>
<li>Number of reads assigned directly to this taxon</li>
<li>A rank code, indicating (U)nclassified, (D)omain, (K)ingdom, (P)hylum, (C)lass, (O)rder, (F)amily, (G)enus, or (S)pecies. All other ranks are simply “-“.</li>
<li>NCBI Taxonomy ID</li>
<li>The indented scientific name</li>
</ol>
</div>
</div>
</div>
<div class="section" id="visualisation-krona">
<h2>6.6. Visualisation (Krona)<a class="headerlink" href="#visualisation-krona" title="Permalink to this headline">¶</a></h2>
<p>We use the <a class="reference external" href="https://github.com/marbl/Krona/wiki">Krona</a> tools to create a nice interactive visualisation of the taxa content of our sample <a class="reference internal" href="#ondov2011" id="id10">[ONDOV2011]</a>.
<a class="reference internal" href="#fig-krona"><span class="std std-numref">Fig. 6.2</span></a> shows an example (albeit an artificial one) snapshot of the visualisation <a class="reference external" href="https://github.com/marbl/Krona/wiki">Krona</a> provides.
<a class="reference internal" href="#fig-krona"><span class="std std-numref">Fig. 6.2</span></a> is a snapshot of the <a class="reference external" href="../_static/taxonomy.krona.html">interactive web-page</a> similar to the one we try to create.</p>
<div class="figure" id="id13">
<span id="fig-krona"></span><img alt="../_images/krona.png" src="../_images/krona.png" />
<p class="caption"><span class="caption-number">Fig. 6.2 </span><span class="caption-text">Example of an Krona output webpage.</span></p>
</div>
<div class="section" id="id11">
<h3>6.6.1. Installation<a class="headerlink" href="#id11" title="Permalink to this headline">¶</a></h3>
<p>Install <a class="reference external" href="https://github.com/marbl/Krona/wiki">Krona</a> with:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="nb">source</span> activate ngs
conda install krona
</pre></div>
</div>
<p>First some house-keeping to make the <a class="reference external" href="https://github.com/marbl/Krona/wiki">Krona</a> installation work.
Do not worry to much about what is happening here.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1"># we delete a symbolic link that is not correct</span>
rm -rf ~/miniconda3/envs/ngs/opt/krona/taxonomy

<span class="c1"># we create a directory in our home where the krona database will live</span>
mkdir -p ~/krona/taxonomy

<span class="c1"># now we make a symbolic link to that directory</span>
ln -s ~/krona/taxonomy ~/miniconda3/envs/ngs/opt/krona/taxonomy
</pre></div>
</div>
</div>
<div class="section" id="build-the-taxonomy">
<h3>6.6.2. Build the taxonomy<a class="headerlink" href="#build-the-taxonomy" title="Permalink to this headline">¶</a></h3>
<p>We need to build a taxonomy database for <a class="reference external" href="https://github.com/marbl/Krona/wiki">Krona</a>.
However, if this fails we will skip this step and just download a pre-build one.
Lets first try to build one.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>ktUpdateTaxonomy.sh ~/krona/taxonomy
</pre></div>
</div>
<p>Now, if this fails, we download a pre-build taxonomy database for krona:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="c1"># Download pre-build database</span>
curl -O http://compbio.massey.ac.nz/data/taxonomy.tab.gz

<span class="c1"># we unzip the file</span>
gzip -d taxonomy.tab.gz

<span class="c1"># we move the unzipped file to our taxonomy directory we specified in the step before.</span>
mv taxonomy.tab ~/krona/taxonomy
</pre></div>
</div>
<div class="admonition attention">
<p class="first admonition-title">Attention</p>
<p class="last">Should this also fail we can download a pre-build database on the <a class="reference internal" href="../downloads.html#downloads"><span class="std std-ref">Downloads</span></a> page via a browser.</p>
</div>
</div>
<div class="section" id="visualise">
<h3>6.6.3. Visualise<a class="headerlink" href="#visualise" title="Permalink to this headline">¶</a></h3>
<p>Now, we use the tool <code class="docutils literal notranslate"><span class="pre">ktImportTaxonomy</span></code> from the <a class="reference external" href="https://github.com/marbl/Krona/wiki">Krona</a> tools to create the html web-page.
We first need build a two column file (<code class="docutils literal notranslate"><span class="pre">read_id&lt;tab&gt;tax_id</span></code>) as input to the <code class="docutils literal notranslate"><span class="pre">ktImportTaxonomy</span></code> tool.
We will do this by cutting the columns out of either the <a class="reference external" href="https://www.ccb.jhu.edu/software/kraken2/">Kraken2</a> or <a class="reference external" href="http://www.ccb.jhu.edu/software/centrifuge/index.shtml">Centrifuge</a> results:</p>
<div class="code bash highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># Kraken2</span>
<span class="n">cd</span> <span class="n">kraken</span>
<span class="n">cat</span> <span class="n">evolved</span><span class="o">-</span><span class="mf">6.</span><span class="n">kraken</span> <span class="o">|</span> <span class="n">cut</span> <span class="o">-</span><span class="n">f</span> <span class="mi">2</span><span class="p">,</span><span class="mi">3</span> <span class="o">&gt;</span> <span class="n">evolved</span><span class="o">-</span><span class="mf">6.</span><span class="n">kraken</span><span class="o">.</span><span class="n">krona</span>
<span class="n">ktImportTaxonomy</span> <span class="n">evolved</span><span class="o">-</span><span class="mf">6.</span><span class="n">kraken</span><span class="o">.</span><span class="n">krona</span>
<span class="n">firefox</span> <span class="n">taxonomy</span><span class="o">.</span><span class="n">krona</span><span class="o">.</span><span class="n">html</span>

<span class="c1"># Centrifuge</span>
<span class="n">cd</span> <span class="n">centrifuge</span>
<span class="n">cat</span> <span class="n">evolved</span><span class="o">-</span><span class="mi">6</span><span class="o">-</span><span class="n">R1</span><span class="o">-</span><span class="n">results</span><span class="o">.</span><span class="n">txt</span> <span class="o">|</span> <span class="n">cut</span> <span class="o">-</span><span class="n">f</span> <span class="mi">1</span><span class="p">,</span><span class="mi">3</span> <span class="o">&gt;</span> <span class="n">evolved</span><span class="o">-</span><span class="mi">6</span><span class="o">-</span><span class="n">R1</span><span class="o">-</span><span class="n">results</span><span class="o">.</span><span class="n">krona</span>
<span class="n">ktImportTaxonomy</span> <span class="n">evolved</span><span class="o">-</span><span class="mi">6</span><span class="o">-</span><span class="n">R1</span><span class="o">-</span><span class="n">results</span><span class="o">.</span><span class="n">krona</span>
<span class="n">firefox</span> <span class="n">taxonomy</span><span class="o">.</span><span class="n">krona</span><span class="o">.</span><span class="n">html</span>
</pre></div>
</div>
<p>What happens here is that we extract the second and third column from the <a class="reference external" href="https://www.ccb.jhu.edu/software/kraken2/">Kraken2</a> results.
Afterwards, we input these to the <a class="reference external" href="https://github.com/marbl/Krona/wiki">Krona</a> script, and open the resulting web-page in a bowser.
Done!</p>
<p class="rubric">References</p>
<table class="docutils citation" frame="void" id="kim2017" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id6">[KIM2017]</a></td><td>Kim D, Song L, Breitwieser FP, Salzberg SL. Centrifuge: rapid and
sensitive classification of metagenomic sequences. <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/27852649">Genome Res. 2016 Dec;26(12):1721-1729</a></td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="lu2017" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id3">[LU2017]</a></td><td>Lu J, Breitwieser FP, Thielen P, Salzberg SL. Bracken: estimating species abundance in metagenomics data. <a class="reference external" href="https://peerj.com/articles/cs-104/">PeerJ Computer Science, 2017, 3:e104, doi:10.7717/peerj-cs.104</a></td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="ondov2011" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id10">[ONDOV2011]</a></td><td>Ondov BD, Bergman NH, and Phillippy AM. Interactive metagenomic visualization in a Web browser. <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/21961884">BMC Bioinformatics, 2011, 12(1):385.</a></td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="wood2014" rules="none">
<colgroup><col class="label" /><col /></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id2">[WOOD2014]</a></td><td>Wood DE and Steven L Salzberg SL. Kraken: ultrafast metagenomic sequence classification using exact alignments. <a class="reference external" href="http://doi.org/10.1186/gb-2014-15-3-r46">Genome Biology, 2014, 15:R46. DOI: 10.1186/gb-2014-15-3-r46</a>.</td></tr>
</tbody>
</table>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../ngs-variantcalling/index.html" class="btn btn-neutral float-right" title="7. Variant calling" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="../ngs-mapping/index.html" class="btn btn-neutral" title="5. Read mapping" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2016-2019, Sebastian Schmeier (https://sschmeier.com).
      Last updated on May 26, 2020.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'../',
            VERSION:'2019.03',
            LANGUAGE:'None',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true,
            SOURCELINK_SUFFIX: '.txt'
        };
    </script>
      <script type="text/javascript" src="../_static/jquery.js"></script>
      <script type="text/javascript" src="../_static/underscore.js"></script>
      <script type="text/javascript" src="../_static/doctools.js"></script>
      <script type="text/javascript" src="../_static/language_data.js"></script>
      <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>

  

  <script type="text/javascript" src="../_static/js/theme.js"></script>

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>